examples/sync/src/bin/client.rs

```diff
@@ -30,66 +34,97 @@ struct Config {
     storage_dir: String,
     /// Port on which metrics are exposed.
     metrics_port: u16,
+    /// Interval for requesting target updates.
+    target_update_interval: Duration,
 }
 
-#[derive(Debug)]
-struct ServerMetadata {
-    root: Digest,
-    oldest_retained_loc: u64,
-    latest_op_loc: u64,
-}
-
-/// Get server metadata to determine sync parameters.
-async fn get_server_metadata<E>(
-    resolver: &Resolver<E>,
-) -> Result<ServerMetadata, Box<dyn std::error::Error>>
+/// Periodically request target updates from server and send them to sync client
+/// on `update_sender`.
+async fn target_update_task<E>(
+    context: E,
+    resolver: Resolver<E>,
+    update_sender: mpsc::Sender<SyncTarget<Digest>>,
+    interval_duration: Duration,
+    initial_target: SyncTarget<Digest>,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync>>
 where
-    E: commonware_runtime::Network + Clone,
+    E: commonware_runtime::Network + commonware_runtime::Clock + Clone,
 {
-    info!("Requesting server metadata...");
+    let mut current_target = initial_target;
 
-    let metadata = resolver.get_server_metadata().await?;
-    let metadata = ServerMetadata {
-        root: metadata.root,
-        oldest_retained_loc: metadata.oldest_retained_loc,
-        latest_op_loc: metadata.latest_op_loc,
-    };
-    info!(?metadata, "Received server metadata");
-    Ok(metadata)
+    loop {
+        context.sleep(interval_duration).await;
+
+        // Request sync target from server
+        match resolver.get_sync_target().await {
+            Ok(new_target) => {
+                // Check if target has changed
+                if new_target.root != current_target.root {
+                    // Send new target to sync client
+                    if let Err(e) = update_sender.clone().try_send(new_target.clone()) {
+                        warn!(error = %e, "failed to send target update to sync client");
+                    } else {
+                        info!(
+                            old_target = ?current_target,
+                            new_target = ?new_target,
+                            "target updated"
+                        );
+                        current_target = new_target;
+                    }
+                } else {
+                    debug!("sync target unchanged");
```

Comment:
nit:
* put `current_target` here (like you did above)
* use `target unchanged` for consistency